packages/wallets/src/evm/connectors/smart-wallet/lib/erc4337-signer.ts

```diff
@@ -43,6 +45,19 @@ export class ERC4337EthersSigner extends Signer {
     transaction: utils.Deferrable<providers.TransactionRequest>,
     options?: UserOpOptions,
   ): Promise<providers.TransactionResponse> {
+    if (!this.approving) {
```

Comment:
do we actually need this? what is this guarding against

Reply:
guarding against duplicate initcode mainly or tx spam ending up approving multiple times

Reply:
hm yeah cause you're doing a recursive call in there. ok that's fine